src/Components/Components/ref/Microsoft.AspNetCore.Components.netstandard2.0.cs

```diff
@@ -247,6 +237,20 @@ public partial class NavigationException : System.Exception
         public NavigationException(string uri) { }
         public string Location { [System.Runtime.CompilerServices.CompilerGeneratedAttribute]get { throw null; } }
     }
+    public abstract partial class NavigationManager
```

Comment:
I think a better name would be Location. There is a well-known browser abstraction that this class maps to (in its own way) which is the browser location. In my opinion we would be better off if we name it Location or BrowserLocation.

Thoughts?

Reply:
@SteveSandersonMS  any thoughts about these suggestions? I like `NavigationManager` as well as I like `Location` or `LocationManager` - so I plan to leave it as-is.

@javiercn - for context we want to avoid naming something **exactly** the same as a DOM/Browser concept because we don't intend for it to expose the same API surface. Location has lots of stuff https://developer.mozilla.org/en-US/docs/Web/API/Location

Reply:
Most of the stuff in Location is exposed as part of System.URI, there is functionality in location that exists for good reason and providing an abstraction for it that is similar but not equal is problematic too.

For example, this abstraction doesn't allow you to navigate while replacing the history entry in the browser.

The problem is, if now, to support that I introduce another abstraction, I have two abstractions to fundamentally do the same thing in my app.

Just a thought.

Reply:
The JS history APIs are pretty awful; I wouldn't be keen to replicate them 1:1.

> For example, this abstraction doesn't allow you to navigate while replacing the history entry in the browser. The problem is, if now, to support that I introduce another abstraction, I have two abstractions to fundamentally do the same thing in my app.

This is a good thing to consider. I don't know that we have any reasonable solution to that though. If we want to add any new navigation primitives in the future, it's necessarily a breaking change for someone who has subclassed `NavigationManager`. I mean, we can add new overloads where the base implementation throws `NotImplementedException` so it's not technically breaking, but from a usefulness perspective it is.

> any thoughts about these suggestions? I like NavigationManager as well as I like Location or LocationManager - so I plan to leave it as-is.

I'd be fine with the name `LocationManager` too. TBH it doesn't feel any different to me.

@javiercn's points about the similarities and differences with browser APIs are very valid, and the same ones we considered when discussing whether to name it `Navigator` or not. I think it's unlikely that we're about to replicate the JS primitives in .NET exactly, at least not in *this* abstraction, especially given the intent for `M.A.Components` to be somewhat web-agnostic. I'd imagine the Mono suite of JS/DOM .NET wrappers would offer direct access to the underlying JS primitives for people who want that. There's probably no single perfect solution - we're trading off competing objectives here, and I'm fairly happy with the position we've got to as of this PR.

Reply:
I don't feel strongly motivated to change this to `LocationManager`. I think both are pretty clear.